metadata-ingestion/src/datahub/utilities/sqlglot_lineage.py

```diff
@@ -276,12 +295,13 @@ def get_urn_for_table(self, table: _TableName, lower: bool = False) -> str:
     def resolve_table(self, table: _TableName) -> Tuple[str, Optional[SchemaInfo]]:
         urn = self.get_urn_for_table(table)
 
-        schema_info = self._resolve_schema_info(urn)
-        if schema_info:
-            return urn, schema_info
+        if not (self.urns and urn not in self.urns):
```

Comment:
I don't really like this logic. Was thinking about creating a "top" set that overrides `__contains__` to always return True and setting that as the default instead of `None`, but could see how that could be confusing. Thoughts? This type of logic comes up in the sql parsing builder too